include/aspect/initial_topography_model/interface.h

```diff
@@ -0,0 +1,170 @@
+/*
+  Copyright (C) 2011, 2012, 2014, 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__initial_topography_model_interface_h
+#define __aspect__initial_topography_model_interface_h
+
+#include <aspect/plugins.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/point.h>
+
+#include <set>
+
+
+namespace aspect
+{
+  /**
+   * A namespace for the definition of properties of the initial topography.
+   * This includes mainly the storage and retrival of the initial topography.
+   * The retrival is done through the value function, which requires a point
+   * of size dim-1, and it return a double which represents the elevation.
+   *
+   * @ingroup InitialTopographyModels
+   */
+  namespace InitialTopographyModel
+  {
+    using namespace dealii;
+
+    /**
+     * Base class for classes that describe particular initial topographies
+     * for the domain.
+     *
+     * @ingroup InitialTopographyModels
+     */
+    template <int dim>
+    class Interface
+    {
+      public:
+        /**
+         * Destructor. Made virtual to enforce that derived classes also have
+         * virtual destructors.
+         */
+        virtual ~Interface();
+
+        /**
+         * Initialization function. This function is called once at the
+         * beginning of the program after parse_parameters is run and after
+         * the SimulatorAccess (if applicable) is initialized.
+         */
+        virtual void initialize ();
+
+        /**
+         * Return the value of the elevation at the given point.
+         */
+        virtual
+        double value (const Point<dim-1> &p) const = 0;
```

Comment:
Because when you call the function you call topograph.value(point), which makes more sense thatn topography.elevation(point).